spec/wings/value_mapper_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+require 'spec_helper'
+require 'wings/value_mapper'
+
+RSpec.describe Wings::ValueMapper do
+  subject(:mapper) { described_class.for(value) }
+  let(:value)      { 'a value' }
+  let(:uri)        { RDF::URI('http://example.com/moomin') }
+
+  describe '.for' do
+    it 'returns a value mapper' do
+      expect(described_class.for(value)).to be_a described_class
+    end
+  end
+
+  describe '.result' do
+    it 'returns the value by default' do
+      expect(mapper.result).to eq value
+    end
+
+    context 'with an ActiveTriples::RDFSource URI value' do
+      let(:value) { ActiveTriples::Resource.new(uri) }
+
+      it 'casts to the RDF term' do
+        expect(mapper.result).to eq uri
+      end
+    end
+
+    context 'with an ActiveTriples::RDFSource bnode value' do
+      let(:value) { ActiveTriples::Resource.new(node) }
+      let(:node)  { RDF::Node.new }
+
+      it 'somehow manages to map the node?'
+    end
+
+    context 'with an ActiveTriples::Relation value' do
+      let(:value)       { work.source }
+      let(:work)        { GenericWork.new(source: cast_values) }
+      let(:cast_values) { ['moomin', 1.125, :moomin, Time.now.utc, false, uri] }
+
+      it 'casts internal values' do
+        expect(mapper.result).to contain_exactly(*cast_values)
+      end
+    end
+
+    context 'with a URI value'
+    context 'with a blank node value'
+    context 'with a boolean value'
+    context 'with a date value'
+    context 'with a numeric value'
```

Comment:
Is it possible to fill out these contexts prior to merge?

Reply:
i'm not sure they all will pass, presently. these are added mostly to catalogue things we need to check up on. at the very least, i expect blank node handling to be unsatisfying as things stand.